### PR TITLE
Separate default validator Validity Predicate

### LINKED ISSUE #25: default validator VP
related to #47 

Add a default validator VP which should be used for the genesis validators. We still need to hash out a bit more details on how we'll reuse code in wasm, for now any logic (which is minimal) can be in-lined into the source.

Steps

* [ ] add a validator VP source (can just be a new directory copied from `vp_template`)
* [ ] store validator's tendermint public key in storage
* [ ] in the VP, disallow to change the public key (this is temporary, later we can allow to change the key when authorized)
* [ ] attach the built wasm to genesis validator(s)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201757896515792/1201762999277367) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by batconjurer**
closes anoma/namada#3 

This PR adds a new vp specifically for validators. Previously validators used the generic user vp. This new vp is added to all newly created validator accounts as well as validators created at genesis. 

The vp adds the following extra functionality: 

* Checks signatures against a validator's user or protocol key depending on the tx type (TODO).
* TODO

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201757896515792/1201763324337448) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by batconjurer** [comment]
pls update wasm

**Comment by batconjurer** [comment]
pls update ci

**Comment by batconjurer** [comment]
pls update wasm

**Comment by batconjurer** [comment]
pls update ci

File: genesis/e2e-tests-single-node.toml [hunk 1]
```diff
@@ -129,7 +129,14 @@ vp = "vp_user"
 # filename (relative to wasm path used by the node)
 filename = "vp_user.wasm"
 # SHA-256 hash of the wasm file
-sha256 = "dc7b97f0448f2369bd2401c3c1d8898f53cac8c464a8c1b1f7f81415a658625d"
+sha256 = "db02f00f1817f44db68a3dd3eaa42f8c301f892a708e7d7aa99c4fea9887f1a7"
+
+# Default validator VP
+[wasm.vp_user]
```
Comment (tzemanovic) [inline genesis/e2e-tests-single-node.toml:135]:
```suggestion
[wasm.vp_validator]
```

File: apps/src/lib/config/genesis.rs [hunk 2]
```diff
@@ -622,7 +623,7 @@ pub fn genesis() -> Genesis {
         account_key: account_keypair.public,
         non_staked_balance: token::Amount::whole(100_000),
         // TODO replace with https://github.com/anoma/anoma/issues/25)
```
Comment (tzemanovic) [inline apps/src/lib/config/genesis.rs:625]:
```suggestion
```

File: wasm/wasm_source/src/vp_validator.rs [hunk 3]
```diff
@@ -0,0 +1,590 @@
+//! A basic user VP.
```
Comment (tzemanovic) [inline wasm/wasm_source/src/vp_validator.rs:1]:
```suggestion
//! A default validator VP.
```

**Pull request closed without merging**